src/libs/auth.js

```diff
@@ -19,29 +20,94 @@ import {
 import * as Utils from 'src/libs/utils'
 
 
+export const getOidcConfig = () => {
+  const metadata = {
+    authorization_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
+    token_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/token`,
+    ...(isGoogleAuthority() && {
+      userinfo_endpoint: 'https://openidconnect.googleapis.com/v1/userinfo',
+      revocation_endpoint: 'https://oauth2.googleapis.com/revoke'
+    })
+  }
+  return {
+    authority: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
+    client_id: authStore.get().oidcConfig.clientId,
+    popup_redirect_uri: `${window.origin}/redirect-from-oauth`,
+    silent_redirect_uri: `${window.origin}/redirect-from-oauth-silent`,
+    metadata,
+    prompt: 'consent login',
+    scope: 'openid email profile',
+    loadUserInfo: isGoogleAuthority(),
+    userStore: new WebStorageStateStore({ store: window.localStorage }),
+    automaticSilentRenew: true,
+    includeIdTokenInSilentRenew: true,
+    extraQueryParams: { access_type: 'offline' },
+    revokeTokenTypes: ['access_token', 'refresh_token']
+  }
+}
+
+const isGoogleAuthority = () => {
+  return _.startsWith('https://accounts.google.com', authStore.get().oidcConfig.authorityEndpoint)
+}
+
 const getAuthInstance = () => {
-  return window.gapi.auth2.getAuthInstance()
+  return authStore.get().authContext
 }
 
 export const signOut = () => {
   // Note: Ideally, we'd actually clear the Leo proxy cookie here, but there's not currently an endpoint to do that.
   // When IA-2236 is done, add that call here.
   cookieReadyStore.reset()
   sessionStorage.clear()
-  getAuthInstance().signOut()
+  const auth = getAuthInstance()
+  auth.revokeTokens()
+    .catch(() => false)
+    .then(() => auth.removeUser())
+    .then(() => auth.clearStaleState())
```

Comment:
Would it be desirable to call `clearStaleState()` if `removeUser()` fails?

```suggestion
    .finally(() => auth.clearStaleState())
```